### ISSUE TITLE
Find My Own way: TLB Mapping Errors.

### ISSUE THREAD

**Issue by ghost** [PCSX2/pcsx2#3984]
**Describe the bug**

The Find my own way tech demo throws an insane number of TLB misses related to scratchpad memory errors.

![image](https://user-images.githubusercontent.com/26351275/101393793-e8c7fd00-38c7-11eb-8cac-7a8255978d4a.png)

Note: the demo still continue to run despite the TLB misses but it is completely broken in every aspect:

![image](https://user-images.githubusercontent.com/26351275/101393902-157c1480-38c8-11eb-84f4-5d48650e9764.png)

**To Reproduce**

Run the Find my own way tech demo until it massively break.

**Expected behavior**

The demo should just works fine like is shown on the screenshot of a real PS2:

![image](https://user-images.githubusercontent.com/26351275/101394019-3b091e00-38c8-11eb-8cdf-68233ab13615.png)

**Emulation Settings**

Everything stock aside small changes to the cyclerate for some other scenes to work better.

**System Info:**

PCSX2 revision: pcsx2-v1.7.0-dev-487-g6ad8563c1-windows-x86
OS: Windows 10
CPU: I9 9900K
GPU: GTX 1080

**Comment by Mrlinkwii** [comment]
@StillPlay2  the issue seems to be a coree issue not a GS one , so i dont think a gs dump will be helpful :)

**Comment by Goatman13** [comment]
Demo map "Virtual Scratchpad" in TLB.
![Nowy Obraz JPEG](https://github.com/PCSX2/pcsx2/assets/101417270/58b00e24-fe59-41b8-903f-b131cc513c79)

**Comment by refractionpcsx2** [comment]
TLB huh? We're boned then :D

**Comment by Goatman13** [comment]
Small patch to workaround this issue. Not sure if it's worth adding to GameDB, i didn't tested it too much.
Demo additionally need VU0 run behind to fix sps here and there, full sync not work. 
```
// Rearrange code to use EE memory instead of (V)Scratchpad
patch=1,EE,E002B488,extended,001076A8
patch=1,EE,20107738,extended,3C0F01F0
patch=1,EE,20107E18,extended,3C0101F0
```
![Nowy Obraz JPEG (3)](https://github.com/PCSX2/pcsx2/assets/101417270/47fb7bba-9c9f-46d0-a640-f7fc1f5b9f32)

**Comment by F0bes** [comment]
I've copied the exact TLB entry (number 0x27) for virtual page 0x70004000.
I also included the preceding write to entry number 47 (all zeros) and I cannot get this to work as you would expect on my PS2.
There is no TLB or bus exception when writing to 0x70004xxx but when you read from this virtual address it's all zeros.
It has nothing to do with the virtual address or caching modes, it's something to do with the physical address in EntryLo0 and EntryLo1.

| Register | Value | PFN | Shifted by the TLB to Form a Physical Address |
|-----------|-------------|----------|-------------|
| EntryLo0 | 0x01c0011f | 0x70004 | 0x70004000
| EntryLo1 | 0x01c0015f | 0x70005 | 0x70005000

Raw PCSX2 Log: `MAP TLB 39: 0x70004000-> [0x70004000 0x70005000] S=0 G=1 ASID=0 Mask=0x000 EntryLo0 PFN=70004 EntryLo0 Cache=3 EntryLo1 PFN=70005 EntryLo1 Cache=3 VPN2=70004000`

After examination, it appears that they messed up and put the same virtual address for the physical address? This of course wouldn't work because 0x70004000 is well out of range for the physical memory the PS2 has. Why isn't this a bus error? I'm not sure why but we should probably do the same and return `0` on reads and ignore on writes.

**Comment by Goatman13** [comment]
> It has nothing to do with the virtual address or caching modes, it's something to do with the physical address in EntryLo0 and EntryLo1.

I think it is cache related. Data cache is setup right after that TLB map, and it is using 0x70004000 as physical address in tags. As far as i know EE don't really check entries for cache, as long as TLB say address is fine, address is fine when used later. Of course, non-existing memory return 0 and write nothing on ps2 (for example bios maps 0x2000000+, games developed by In Utero rely on that due to optimistic stack usage).

What if you map 0x70004000 and hint cache to use it? My theory is that demo use data cache in very clever way, as additional memory. While obviously you can't RW 0x70004000 even if you map it as valid, cache is different story.

**Comment by F0bes** [comment]
> > It has nothing to do with the virtual address or caching modes, it's something to do with the physical address in EntryLo0 and EntryLo1.
> 
> I think it is cache related. Data cache is setup right after that TLB map, and it is using 0x70004000 as physical address in tags. As far as i know EE don't really check entries for cache, as long as TLB say address is fine, address is fine when used later. Of course, non-existing memory return 0 and write nothing on ps2 (for example bios maps 0x2000000+, games developed by In Utero rely on that due to optimistic stack usage).
> 
> What if you map 0x70004000 and hint cache to use it? My theory is that demo use data cache in very clever way, as additional memory. While obviously you can't RW 0x70004000 even if you map it as valid, cache is different story.

I didn't see your comment until today. I spent yesterday working this out, and yes, you are right. 
I've implemented this in #3984 but this is one of those games that sometimes deadlock with ee cache enabled (before the bus errors come in).

**Comment by F0bes** [comment]
Although dreadfully slow with EE Cache, #11459 fixes this demo.
